reactor-netty-http/src/main/java/reactor/netty/http/server/HttpServerFormDecoderProvider.java

```diff
@@ -0,0 +1,403 @@
+		 * <p>Note:
+		 * <ul>
+		 *     <li>If set to {@code -1} the entire contents is stored in memory</li>
+		 *     <li>If set to {@code 0} the entire contents is stored on disk</li>
+		 * </ul>
+		 *
+		 * @param maxInMemorySize the maximum in-memory size
+		 * @return {@code this}
+		 */
+		Builder maxInMemorySize(long maxInMemorySize);
+
+		/**
+		 * Set the maximum size per {@link Attribute}/{@link FileUpload}. When the limit is reached, an exception is raised.
+		 * Default to {@link DefaultHttpDataFactory#MAXSIZE} - unlimited.
+		 * <p>Note: If set to {@code -1} this means no limitation.
+		 *
+		 * @param maxSize the maximum in-memory size
+		 * @return {@code this}
+		 */
+		Builder maxSize(long maxSize);
+
+		/**
+		 * Sets the scheduler to be used for offloading disk operations in the decoding phase.
+		 * Default to {@link Schedulers#boundedElastic()}
+		 *
+		 * @param scheduler the scheduler to be used for offloading disk operations in the decoding phase
+		 * @return {@code this}
+		 */
+		Builder scheduler(Scheduler scheduler);
+
+		/**
+		 * When set to {@code true}, the data is streamed directly from the parsed input buffer stream,
+		 * which means it is not stored either in memory or file.
+		 * When {@code false}, parts are backed by in-memory and/or file storage. Default to {@code false}.
+		 * <p><strong>NOTE</strong> that with streaming enabled, the provided {@link Attribute}/{@link FileUpload}
+		 * might not be in a complete state i.e. {@link HttpData#isCompleted()} has to be checked.
+		 * <p>Also note that enabling this property effectively ignores
+		 * {@link #maxInMemorySize(long)},
+		 * {@link #baseDirectory(Path)}, and
+		 * {@link #scheduler(Scheduler)}.
+		 */
+		Builder streaming(boolean enable);
+	}
+
+	final Path baseDirectory;
+	final Charset charset;
+	final long maxInMemorySize;
+	final long maxSize;
+	final Scheduler scheduler;
+	final boolean streaming;
+
+	HttpServerFormDecoderProvider(Build build) {
+		this.baseDirectory = build.baseDirectory;
+		this.charset = build.charset;
+		this.maxInMemorySize = !build.streaming ? build.maxInMemorySize : -1;
+		this.maxSize = build.maxSize;
+		this.scheduler = build.scheduler;
+		this.streaming = build.streaming;
+	}
+
+	/**
+	 * Returns the configured directory where to store disk {@link Attribute}/{@link FileUpload}.
+	 *
+	 * @return the configured directory where to store disk {@link Attribute}/{@link FileUpload}
+	 */
+	@Nullable
+	public Path baseDirectory() {
+		return baseDirectory;
+	}
+
+	/**
+	 * Returns the configured charset for {@link Attribute}/{@link FileUpload}.
+	 *
+	 * @return the configured charset for {@link Attribute}/{@link FileUpload}
+	 */
+	public Charset charset() {
+		return charset;
+	}
+
+	/**
+	 * Returns the configured maximum in-memory size per {@link Attribute}/{@link FileUpload}.
+	 *
+	 * @return the configured maximum in-memory size per {@link Attribute}/{@link FileUpload}
```

Comment:
```suggestion
	 * Returns the configured maximum size after which an {@link Attribute}/{@link FileUpload} starts being stored on disk rather than in memory.
	 *
	 * @return the configured maximum size after which an {@link Attribute}/{@link FileUpload} starts being stored on disk rather than in memory
```